### ISSUE TITLE
Fix full text search functionality

### ISSUE THREAD

**Issue by EeveesEyes** [hochschule-darmstadt-UAS/ddk-artbrowser#51]
**Describe the bug**
Full text search for entities does not result in artworks containing those entities.

**To Reproduce**
1. Go to https://cai-ddk-art-browser-staging.fbi.h-da.de
2. Search for a genre/location/artist/... eg. 'Buchseite'
3. See no results

**Expected behavior**
Artworks with foreign keys of 'Buchseite' should be found.

**Comment by BernhardHumm** [comment]
“Buchseite” is not found since this text is not contained in any attribute of the artwork but in a related entity (type).

This is technically valid (insofar not really a bug) but not intuitive from the users’ point of view.

 

There is already a feature issue for enhancing the full-text search which includes this issue.

I suggest not to treat this as a bug (there will be no simple fix) but to solve it as part of the feature issue mentioned.

 

 

Von: Adrian Kailus <notifications@github.com> 
Gesendet: Donnerstag, 17. Dezember 2020 13:08
An: hochschule-darmstadt-UAS/ddk-artbrowser <ddk-artbrowser@noreply.github.com>
Cc: Subscribed <subscribed@noreply.github.com>
Betreff: [hochschule-darmstadt-UAS/ddk-artbrowser] Fix full text search functionality (#51)

 

Describe the bug
Full text search for entities does not result in artworks containing those entities.

To Reproduce
Steps to reproduce the behavior:

1.	Go to https://cai-ddk-art-browser-staging.fbi.h-da.de
2.	Search for a genre/location/artist/... eg. 'Buchseite'
3.	See no results

Expected behavior
Artworks with foreign keys of 'Buchseite' should be found.

**Comment by EeveesEyes** [comment]
Actually the [existing issue](https://github.com/hochschule-darmstadt/openartbrowser/issues/199) is about a complete redesign of the search-result component rather than the modification of the result set. Furthermore this issue is not ready yet. Therefore this issue is much smaller.
The solution i think of should be no more than two queries: The first one searches all documents for the term (we already have this for artworks only, so this is no additional request) - the result set is then split by type and the second query searches artworks with those related ids. Actually we have these queries as well, but they are executed first. A simple reordering of the queries should do the trick.
Therefore i would treat this issue not as a bug but as an "usability bug". The feature-part would be the new result page.

**Comment by EeveesEyes** [comment]
Do you agree with doing this modification first and doing the (fulltext) search result page design later? I think both issues are not necessarily dependent on each other.

**Comment by BernhardHumm** [comment]
No, the full-text search should not be implemented via several queries. To implement the desired functionality, the index needs to be extended.
